### PR TITLE
fix: polyfill ResizeObserver

### PR DESCRIPTION
**Pull request by vm**
## Description
https://uniswap-labs.sentry.io/issues/4178534607/?project=4504255148851200&referrer=release-issue-stream

ResizeObserver is not polyfilled by default. This makes sure we do that for legacy browsers (like slightly older Safari mobile). I switched the polyfill library to what's recommended by https://github.com/ZeeCoder/use-resize-observer#transpilation--polyfilling

### Automated testing
- [x] Unit test
- [x] Integration/E2E test

### CODE REVIEW

File: src/polyfills.ts
Comment (zzmp) [inline src/polyfills.ts:3]: Why change our dependency instead of just importing the existing package in `src/polyfills`? What makes `@juggle/resize-observer` preferable to `resize-observer-polyfill`?

NB: `resize-observer-polyfill` has much more usage, although it is larger and staler.

Comment (vm) [reply src/polyfills.ts:3]: it just looked like it was more maintained tbh, but we can leave the old one if better

Comment (cmcewen) [reply src/polyfills.ts:3]: if you look at yarn.lock we already had `@juggle/resize-observer` somewhere - the net of this is just removing a dep

**Review by cmcewen — APPROVED** [review]

File: src/setupTests.ts
Comment (zzmp) [inline src/setupTests.ts:4]: Instead of this, can you just `import 'polyfills'` on L3?

**Review by zzmp — APPROVED** [review]
lgtm with a nit: I think it'd be better to just `import 'polyfills'` in `setupTests.ts`.

File: src/setupTests.ts
Comment (vm) [reply src/setupTests.ts:4]: gonna do in separate PR so I don't need another approval on this before merging

**Pull request merged**
